### DISCUSSION TITLE
Square bracket numbers (lines of asm) doesn't correspond with what is actually generated

### DISCUSSION THREAD

**Discussion by pozix604** [pacak/cargo-show-asm#343] [Q&A] [answered]
I'm confused by the `[##]` when listing functions to generate assembly for. Here is some sample output:

```
$ cargo asm --lib
    Finished `release` profile [optimized] target(s) in 0.00s

Try one of those by name or a sequence number
0 "func0" [51]
1 "func1" [56]
2 "func2" [60]

$ cargo asm --lib 0 | wc
    Finished `release` profile [optimized] target(s) in 0.00s
     26      73     400

$ cargo asm --lib 1 | wc
    Finished `release` profile [optimized] target(s) in 0.00s
     26      75     410

$ cargo asm --lib 2 | wc
    Finished `release` profile [optimized] target(s) in 0.00s
     25      66     373
```

Why does the summary list 51, 56 and 60 lines of assembly, while the actual assembly displayed is 26, 26, and 25?

The first list of numbers differs from the second. Additionally, the first list is increasing, while the second list is actually _decreasing_. `func2` in the summary has the most number of lines, while the actual displayed assembly has the least number of lines.

**Comment by pacak** [comment, marked as answer]
They are calculated from the actual lines llvm produces. Those lines contain a bunch of redundant things that `cargo-show-asm` strips - labels that aren't used (pass `-K` to see those) and some of the debug info - location directives, etc. You can pass `-vvv` to see everything.

I could make it so numbers correspond to what the size that will be printed, but removing redundant labels is not linear in cost and  can slow things down if your file is super chonky.

Main goal for those numbers is to get you a high level overview where most of the code is and to indicate that there can be several functions with the same name.
